web/source/kmwosk.ts

```diff
@@ -702,21 +702,22 @@ if(!window['keyman']['initialized']) {
      */
     osk.getVKDictionaryCode = function(keyName)
     {
-      if(!keymanweb._ActiveKeyboard['VKDictionary'])
+      var activeKeyboard = keymanweb.keyboardManager.activeKeyboard;
+      if(!activeKeyboard['VKDictionary'])
       {
         var a=[];
-        if(typeof keymanweb._ActiveKeyboard['KVKD'] == 'string')
+        if(typeof activeKeyboard['KVKD'] == 'string')
         {
           // Build the VK dictionary
           // TODO: Move the dictionary build into the compiler -- so compiler generates code such as following.  Makes the VKDictionary member unnecessary
```

Comment:
Is this todo comment still required?

Reply:
From our Developer 10-compiled Amharic keyboard:

```
this.KVKD="T_GLOTTAL_E T_PHARYNGEAL_E T_TE T_THE T_GLOTTAL_U T_PHARYNGEAL_U this.KVKD="T_GLOTTAL_E T_PHARYNGEAL_E T_TE T_THE T_GLOTTAL_U T_PHARYNGEAL_U T_GLOTTAL_I T_PHARYNGEAL_I T_GLOTTAL_O T_PHARYNGEAL_O T_PE T_PHE T_GLOTTAL_A T_PHARYNGEAL_A T_SE T_SZE T_TSE T_TZE T_HE T_XE T_HHE T_ZE T_ZHE T_CE T_CHE T_NE T_NYE T_KE T_KXE T_WORDSPACE T_LCHEVRON T_RCHEVRON T_LPARENTHESIS T_LBRKT T_LBRACE T_LANGLE T_RPARENTHESIS T_RBRKT T_RBRACE T_RANGLE T_COMMA_ENG T_COLON_ENG T_SEMICOLON_ENG T_QUESTION_ENG T_EXCLAMATION T_ALPHA T_new_2514 T_new_2515 T_new_2516 T_COMMA T_FULL_STOP T_PARAGRAPH_SEPARATOR T_SECTION_MARK T_QUESTION_MARK T_SEMICOLON T_COLON T_PREFACE_COLON T_GEMINATION_MARK T_ONE T_TWO T_THREE T_FOUR T_FIVE T_SIX T_SEVEN T_EIGHT T_NINE T_TEN T_TWENTY T_THIRTY T_FORTY T_FIFTY T_SIXTY T_SEVENTY T_EIGHTY T_NINETY T_HUNDRED T_TEN_THOUSAND";
```

I'm going with "yes."  The comment indicates a desire to pre-compile this string into a Map<String, Integer> associative array.